src/qt/optionsmodel.cpp

```diff
@@ -151,10 +151,31 @@ void OptionsModel::Init(bool resetSettings)
     language = settings.value("language").toString();
 }
 
+/** Helper function to copy contents from one QSettings to another.
+ * By using allKeys this also covers nested settings in a hierarchy.
+ */
+static void copySettings(QSettings &dst, const QSettings &src)
+{
+    for (const QString &key: src.allKeys()) {
+        dst.setValue(key, src.value(key));
+    }
+}
+
+/** Back up a QSettings to an ini-formatted file. */
+static void backupSettings(const fs::path &filename, const QSettings &src)
+{
+    QSettings dst(GUIUtil::boostPathToQString(filename), QSettings::IniFormat);
+    qInfo() << "Backing up GUI settings to" << GUIUtil::boostPathToQString(filename);
+    copySettings(dst, src);
+}
+
 void OptionsModel::Reset()
 {
     QSettings settings;
 
+    // Backup old settings for troubleshooting
+    backupSettings(GetDataDir(true) / "guisettings.bak", settings);
```

Comment:
nit, `GetDataDir()` argument defaults to `true` so it'd be clearer to leave it out imo

Reply:
Isn't it the other way around: clearer to leave it in, because it's explicit?

Reply:
Normally I'd say yes, but in this case it just leaves me wondering what the argument is doing, because in most other cases of `GetDataDir` in the code, the argument is left out. Personal preference

Reply:
During the initialization sequence it's really important to get this right. E.g. I had to check the order in `src/qt/bitcoin.cpp` that calling `GetDataDir(true)` was allowed at this point.

Agree that boolean arguments are unclear, so are default arguments, so it's kind of choosing between evils here. Instead of an argument to `GetDataDir` we should have done `GetSpecificDataDir` and `GetRootDataDir` or such. Anyhow... I don't think this code is particularly unclear, I'll add 'to chain-specific datadir' in the comment.